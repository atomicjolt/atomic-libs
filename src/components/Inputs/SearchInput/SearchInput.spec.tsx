import React from "react";
import { describe, test, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import { SearchInput } from ".";

describe("matches snapshot", () => {
  test("displays without button", () => {
    const result = render(<SearchInput label="search" onSubmit={() => {}} />);
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("displays with search button", () => {
    const result = render(<SearchInput label="search" onSubmit={() => {}} />);
    expect(result.asFragment()).toMatchSnapshot();
  });
});

// test("Enter calls onSumbit()", () => {
//   const onSubmit = jest.fn(() => {});
//   const result = render(<SearchInput label="search" onSubmit={onSubmit} />);
//   fireEvent.keyPress(screen.getByRole("search"), {
//     key: "Enter",
//     code: "Enter",
//     charCode: 13,
//   });

//   expect(onSubmit.mock.calls.length).toEqual(1);
// });

test("button calls onSumbit()", () => {
  const onSubmit = vi.fn(() => {});
  const result = render(<SearchInput label="label" onSubmit={onSubmit} />);
  fireEvent.click(screen.getByText("search"));

  expect(onSubmit.mock.calls.length).toEqual(1);
});
