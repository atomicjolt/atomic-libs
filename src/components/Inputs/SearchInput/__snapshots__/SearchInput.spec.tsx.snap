// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot > displays with search button 1`] = `
<DocumentFragment>
  <div
    class="sc-cwHqhk bQUpgu aje-input__search"
  >
    <label
      class="sc-imWZod gWbIwJ is-auto"
      for="react-aria-:r5:"
      id="react-aria-:r6:"
    >
      search
    </label>
    <div
      class="sc-dAlxHm cipQZz aje-combo-input padding-left padding-right"
    >
      <input
        aria-labelledby="react-aria-:r6:"
        class="sc-kpDprT fyrTNF"
        id="react-aria-:r5:"
        type="search"
        value=""
      />
      <button
        class="sc-gsFSjX sc-kAycRU fvAPeR joEykT aje-btn aje-btn--icon aje-btn--content is-medium"
        type="button"
      >
        <i
          aria-hidden="true"
          class="sc-iGgVNO cfZcSx aje-icon material-icons is-medium"
        >
          search
        </i>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot > displays without button 1`] = `
<DocumentFragment>
  <div
    class="sc-cwHqhk bQUpgu aje-input__search"
  >
    <label
      class="sc-imWZod gWbIwJ is-auto"
      for="react-aria-:r0:"
      id="react-aria-:r1:"
    >
      search
    </label>
    <div
      class="sc-dAlxHm cipQZz aje-combo-input padding-left padding-right"
    >
      <input
        aria-labelledby="react-aria-:r1:"
        class="sc-kpDprT fyrTNF"
        id="react-aria-:r0:"
        type="search"
        value=""
      />
      <button
        class="sc-gsFSjX sc-kAycRU fvAPeR joEykT aje-btn aje-btn--icon aje-btn--content is-medium"
        type="button"
      >
        <i
          aria-hidden="true"
          class="sc-iGgVNO cfZcSx aje-icon material-icons is-medium"
        >
          search
        </i>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
